src/ol_infrastructure/components/services/k8s.py

```diff
@@ -350,3 +362,367 @@ def __init__(
+        return self
+
+
+class OLApisixRoute(pulumi.ComponentResource):
+    """
+    Route configuration for apisix
+    Defines and creates an "ApisixRoute" resource in the k8s cluster
+    """
+
+    def __init__(
+        self,
+        name: str,
+        route_configs: list[OLApisixRouteConfig],
+        k8s_namespace: str,
+        k8s_labels: dict[str, str],
+        opts: Optional[pulumi.ResourceOptions] = None,
+    ):
+        super().__init__(
+            "ol:infrastructure:services:k8s:OLApisixRoute", name, None, opts
+        )
+
+        resource_options = pulumi.ResourceOptions(parent=self).merge(opts)
+
+        self.apisix_route_resource = kubernetes.apiextensions.CustomResource(
+            f"OLApisixRoute-{name}",
+            api_version="apisix.apache.org/v2",
+            kind="ApisixRoute",
+            metadata=kubernetes.meta.v1.ObjectMetaArgs(
+                name=name,
+                labels=k8s_labels,
+                namespace=k8s_namespace,
+            ),
+            spec={
+                "http": self.__build_route_list(route_configs),
+            },
+            opts=resource_options.merge(
+                pulumi.ResourceOptions(delete_before_replace=True)
+            ),
+        )
+
+    @classmethod
+    def __build_route_list(
+        cls, route_configs: list[OLApisixRouteConfig]
+    ) -> list[dict[str, Any]]:
+        routes = []
+        for route_config in route_configs:
+            route = {
+                "name": route_config.route_name,
+                "priority": route_config.priority,
+                "plugin_config_name": route_config.shared_plugin_config_name,
+                "plugins": [p.model_dump(by_alias=True) for p in route_config.plugins],
+                "match": {
+                    "hosts": route_config.hosts,
+                    "paths": route_config.paths,
+                },
+                "websocket": route_config.websocket,
+            }
+            if route_config.upstream:
+                route["upstreams"] = [{"name": route_config.upstream}]
+            else:
+                route["backends"] = [
+                    {
+                        "serviceName": route_config.backend_service_name,
+                        "servicePort": route_config.backend_service_port,
+                        "resolveGranularity": route_config.backend_resolve_granularity,
+                    }
+                ]
+            routes.append(route)
+        return routes
+
+
+class OLApisixOIDCConfig(BaseModel):
+    application_name: str
+    k8s_labels: dict[str, str] = {}
+    k8s_namespace: str
+    vault_mount: str = "secret-operations"
+    vault_mount_type: str = "kv-v1"
+    vault_path: str
+    vaultauth: str
+
+    oidc_bearer_only: bool = False
+    oidc_introspection_endpoint_auth_method: str = "client_secret_basic"
+    oidc_logout_path: str = "/logout/oidc"
+    oidc_post_logout_redirect_uri: str = "/"
+    oidc_refresh_session_interval: NonNegativeInt = 1800
```

Comment:
Delete this line?

Reply:
Based on this thread: https://mitodl.slack.com/archives/C03K5HYGPT9/p1743009733774169?thread_ts=1742911132.231889&cid=C03K5HYGPT9

Assuming this is where it's assigned.